bin/match_zcat_photo

```diff
@@ -0,0 +1,696 @@
+#!/usr/bin/env python
+
+"""Match data release redshift catalogs against the original photometric
+(target) catalogs.
+
+Sequence of commands for a given release (here, fuji)
+1. Gather targeting photometry from the individual redshift catalogs.
+
+   time match_zcat_photo --reduxdir /global/cfs/cdirs/desi/spectro/redux/fuji -o /global/cscratch1/sd/ioannis/photocatalog/fuji --outprefix fuji --mp 32 --targetphot
+
+   Output files:
+    targetphot-fuji.fits
+    targetphot-zcat-fuji.fits
+    targetphot-missing-fuji.fits
+
+2. Gather Tractor photometry.
+
+   time match_zcat_photo --reduxdir /global/cfs/cdirs/desi/spectro/redux/fuji -o /global/cscratch1/sd/ioannis/photocatalog/fuji --outprefix fuji --mp 32 --tractorphot
+
+   Output files:
+     tractorphot-nside4-hp???-fuji.fits
+
+3. Validate Tractor photometry
+
+   time match_zcat_photo --reduxdir /global/cfs/cdirs/desi/spectro/redux/fuji -o /global/cscratch1/sd/ioannis/photocatalog/fuji --outprefix fuji --validate-tractorphot
+
+   Output files:
+     tractorphot-validate-fuji.fits
+
+4. Gather targeting and Tractor photometry for *potential* targets (going back to the original fiberassign files).
+
+   time match_zcat_photo --reduxdir /global/cfs/cdirs/desi/spectro/redux/fuji -o /global/cscratch1/sd/ioannis/photocatalog/fuji --outprefix fuji --mp 32 --targetphot --potential-targets
+   time match_zcat_photo --reduxdir /global/cfs/cdirs/desi/spectro/redux/fuji -o /global/cscratch1/sd/ioannis/photocatalog/fuji --outprefix fuji --mp 32 --tractorphot --potential-targets
+
+   Output files:
+     targetphot-potential-targets-fuji.fits
+     targetphot-potential-targets-zcat-fuji.fits
+     targetphot-potential-targets-missing-fuji.fits
+     tractorphot-potential-targets-nside4-hp???-fuji.fits
+
+"""
+import os, sys, argparse, pdb
+from glob import glob
+import numpy as np
+import fitsio
+import multiprocessing
+import healpy as hp
+
+import astropy
+from astropy.io import fits
+from astropy.table import Table, vstack, join
+
+from desispec.io import specprod_root
+from desimodel.footprint import radec2pix
+from desitarget.targets import decode_targetid
+from desiutil.brick import brickname
+
+from desiutil.log import get_logger, DEBUG
+log = get_logger()#DEBUG)
+
+desi_root = os.environ.get('DESI_ROOT')
+fiberassign_dir = os.path.join(desi_root, 'target', 'fiberassign', 'tiles', 'trunk')
+
+def _resolve(cat):
```

Comment:
Please add docstrings throughout defining inputs and outputs, even for _hidden functions.  Our future selves will thank you.

Reply:
OK, most of these are done.